Sources/Swift/Integrations/Performance/IO/FileManager+SentryTracing.swift

```diff
@@ -0,0 +1,140 @@
+@_implementationOnly import _SentryPrivate
+import Foundation
+
+/// A ``FileManager`` extension that tracks read and write operations with Sentry.
+///
+/// - Note: Methods provided by this extension reflect the same functionality as the original ``FileManager`` methods, but they track the operation with Sentry.
+public extension FileManager {
+
+    // MARK: - Creating and Deleting Items
+
+    /// Creates a file with the specified content and attributes at the given location, tracking the operation with Sentry.
+    ///
+    /// - Important: Using this method with auto-instrumentation for file operations enabled can lead to duplicate spans on older operating system versions.
+    ///              It is recommended to use either automatic or manual instrumentation. You can disable automatic instrumentation by setting
+    ///              `options.experimental.enableFileManagerSwizzling` to `false` when initializing Sentry.
+    /// - Parameters:
+    ///   - path: The path for the new file.
+    ///   - data: A data object containing the contents of the new file.
+    ///   - attr: A dictionary containing the attributes to associate with the new file.
+    ///           You can use these attributes to set the owner and group numbers, file permissions, and modification date.
+    ///           For a list of keys, see ``FileAttributeKey``. If you specify `nil` for attributes, the file is created with a set of default attributes.
+    /// - Returns: `true` if the operation was successful or if the item already exists, otherwise `false`.
+    /// - Note: See ``FileManager.createFile(atPath:contents:attributes:)`` for more information.
+    func createFileWithSentryTracing(atPath path: String, contents data: Data?, attributes attr: [FileAttributeKey: Any]? = nil) -> Bool {
+        let tracker = SentryFileIOTracker.sharedInstance()
```

Comment:
`h`: This initializes a `SentryFileIOTracker` which stays in the SentryDependencyContainer. So when you call this method before initializing the SDK, the SentryThreadInspector initializes with `nil` options which breaks the in app logic for building stacktraces when calling captureError or captureEvent.

https://github.com/getsentry/sentry-cocoa/blob/0c3e79c5162e4316ac705f8376290c688084dd6a/Sources/Sentry/SentryDependencyContainer.m#L193-L202

Reply:
How do we handle this in other parts of the SDK? Do we always check if the SDK is enabled before accessing the dependency container?

Reply:
Usually, we don't really do that, I believe. We could also call DependencyContainer.reset early in SentrySDK.start to solve this problem other dependencies.

Reply:
In that case it's probably safer to add the isEnabled check before accessing the dependency container with a comment why it's there.

Reply:
Yep, DependencyContainer.reset early in SentrySDK.start would still make sense, but in an extra PR.

Reply:
I added the check to the `SentryFileIOTracker.sharedInstance` and made it nullable